src/main/java/com/epam/test_generator/dto/RawCaseResultDTO.java

```diff
@@ -0,0 +1,80 @@
+package com.epam.test_generator.dto;
+
+import com.epam.test_generator.entities.Status;
+import java.time.Duration;
+import java.util.LinkedHashMap;
+import java.util.List;
+import javax.validation.constraints.NotNull;
+
+public class RawCaseResultDTO {
+
+    @NotNull
+    private Long id;
+
+    @NotNull
+    private Long duration;
+
+    @NotNull
+    private Status status;
+
+    @NotNull
+    private List<RawStepResultDTO> steps;
+
+
+    public RawCaseResultDTO(long id, long duration,
+                            Status status,
+                            List<RawStepResultDTO> steps) {
+        this.id = id;
+        this.duration = duration;
+        this.status = status;
+        this.steps = steps;
+    }
+
+    public RawCaseResultDTO() {
+    }
+
+    public long getId() {
+        return id;
+    }
+
+    public List<RawStepResultDTO> getSteps() {
+        return steps;
+    }
+
+    public long getDuration() {
+        return duration;
+    }
+
+    public Status getStatus() {
+        return status;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Check this method. `!(o instanceof CaseDTO)` is invalid
https://docs.oracle.com/javase/7/docs/api/java/lang/Object.html#equals(java.lang.Object)